### ISSUE TITLE
Blurring issues due to canvas dimension rounding

### ISSUE THREAD

**Issue by bcairns** [google/model-viewer#3554]
### Discussed in https://github.com/google/model-viewer/discussions/3549

<div type='discussions-op-text'>

<sup>Originally posted by **bcairns** June 24, 2022</sup>
We are seeing a (fairly subtle) blurring of the final output, most noticeable on edges, and it seems to be due to non-round pixel dimensions of the canvas.

It looks like this (or a similar) issue has been discussed before:

- https://github.com/google/model-viewer/issues/1503
- https://github.com/google/model-viewer/pull/1506
- https://github.com/google/model-viewer/pull/1542

### Details:

As reported by our client, we observe the rendering doesn't look as sharp as a fixed-size viewer that uses round numbers. (This is using latest Chrome, on a Mac, with current 1.12.0 model-viewer)

We are using model-viewer inside a css grid system, which uses percentages for column widths, leading to non-round pixel values for the container and canvas elements:

![image](https://user-images.githubusercontent.com/638372/175699708-8fe0e3be-f2bf-4658-8044-49fb53f0b490.png)

If we set the parent container to a fixed size (using round pixels values), the blurring issue immediately goes away.

![image](https://user-images.githubusercontent.com/638372/175713247-8169d305-9dc2-48c9-b8f4-8f2fe5e87ca5.png)

An adequate workaround may be to use (via injected stylesheet):

`canvas.show {
    width: auto !important;
    height: auto !important;
}`

This overrides the inline CSS styles and causes the canvas to use the integer width/height attributes for its dimensions (eg `width="691" height="533"`).  This does cause some resizing weirdness however, so it's not ideal.

### tl;dr

It seems to us that the integer canvas width/height attributes (eg `width="691" height="533"`) should be precisely the same for the inline CSS styles for the canvas dimensions.

Otherwise having numbers like (eg) 691.6 and 533.6 can lead to "off-by-1" discrepancy on the canvas dimensions, causing scaling to occur which impacts visual quality.</div>

Please see the linked discussion at top for screenshots and some additional comments.

### Solution

Since three.js renderer uses Math.floor in setSize() (see https://github.com/mrdoob/three.js/blob/dev/src/renderers/WebGLRenderer.js#L415), we should standardize on floor() (over round() or ceil()).

For consistency (see above code screenshots), this can be applied to:

- the canvas width/height attributes
- the canvas inline width/height css styles
- the .container element

**Comment by elalish** [comment]
Fixed by #3555
